dolphin/drivers/dell_emc/vmax/client.py

```diff
@@ -123,3 +123,58 @@ def list_pools(self):
             LOG.error("Failed to get pool metrics from VMAX: {}".format(err))
             raise exception.StorageBackendException(
                 reason='Failed to get pool metrics from VMAX')
+
+    def list_volumes(self, storage_id):
+
+        try:
+            # List all volumes
+            volumes = self.conn.provisioning.get_volume_list(filters={'data_volume': 'false'})
+
+            volume_list = []
+            for volume in volumes:
+                # Get volume details
+                vol = self.conn.provisioning.get_volume(volume)
+
+                total_cap = vol['cap_mb'] * units.Mi
+                used_cap = (total_cap * vol['allocated_percent']) / 100.0
+                free_cap = total_cap - used_cap
+
+                # TODO: Update constants.VolumeStatus to make mapping more precise
+                switcher = {
+                    'Ready': constants.VolumeStatus.AVAILABLE,
+                    'Not Ready': constants.VolumeStatus.ERROR,
+                    'Mixed': constants.VolumeStatus.ERROR,
+                    'Write Disabled': constants.VolumeStatus.ERROR,
+                    'N/A': constants.VolumeStatus.ERROR,
+                }
+                status = switcher.get(vol['status'], constants.VolumeStatus.ERROR)
+
+                v = {
+                    "name": volume,
+                    "storage_id": storage_id,
+                    "description": vol['type'],
+                    "status": status,
+                    "original_id": vol['volumeId'],
+                    "wwn": vol['wwn'],
+                    "storage_type": constants.StorageType.BLOCK,
+                    "total_capacity": int(total_cap),
+                    "used_capacity": int(used_cap),
+                    "free_capacity": int(free_cap),
+                }
+
+                if vol['num_of_storage_groups'] == 1:
```

Comment:
Can a volume be part of multiple SGs?

Reply:
Yes.

Reply:
So we are not taking that in account?

Reply:
Also , lets have the logic to assign volumes which are not in any SG to map default_srp as SRP.
When you create volume you cant assign SRP. When you map in to SG , the volume will be part of SRP which is assigned to that SG.

Reply:
Yes, 'TODO'  comment below, is to handle these cases.

Reply:
please create an issue to track.

Reply:
Done. (https://github.com/katurii/dolphin/issues/74)

Reply:
Updated issue #74 with case for multiple SGs

Reply:
Thanks